core/src/serve_repair.rs

```diff
@@ -170,6 +167,95 @@ struct ServeRepairStats {
     err_id_mismatch: usize,
 }
 
+struct RepairBudget {
+    last_update: Instant,
+    epoch_staked_nodes: Option<Arc<HashMap<Pubkey, u64>>>,
+    node_used: HashMap<Pubkey, /*bytes*/ u64>,
+    staked_remaining: u64,
+    unstaked_remaining: u64,
+    staked_request: u64,
+    unstaked_request: u64,
+}
+
+impl RepairBudget {
+    const INTERVAL_MS: u64 = 1_000;
+    const MAX_BYTES_PER_INTERVAL: u64 = 12_000_000;
+    const MAX_STAKED_BYTES_PER_INTERVAL: u64 = 10_000_000;
+    const MAX_NODE_BYTES_PER_INTERVAL: u64 = 1_000_000;
+
+    fn new(bank_forks: &RwLock<BankForks>) -> Self {
+        let root_bank = bank_forks.read().unwrap().root_bank();
+        let epoch_schedule = root_bank.epoch_schedule();
+        let epoch = epoch_schedule.get_epoch(root_bank.slot());
+        let staked_nodes = root_bank.epoch_staked_nodes(epoch);
+        Self {
+            last_update: Instant::now(),
+            epoch_staked_nodes: staked_nodes,
+            node_used: HashMap::default(),
+            staked_remaining: 0,
+            unstaked_remaining: Self::MAX_BYTES_PER_INTERVAL,
```

Comment:
This is confusing why `staked_remaining` is initialized to zero, and `unstaked_remaining` to `MAX_...`. Then below
```rust
let staked_remaining = self.staked_request.min(Self::MAX_STAKED_BYTES_PER_INTERVAL);
```
Can you please explain what is the intended budget allocation policy here?

Reply:
Changed default for staked node budget to `MAX_STAKED_BYTES_PER_INTERVAL`.